### PR TITLE
d_a_obj_ebomzo mostly matched

### PR DESCRIPTION
**Pull request by waffleoRai**
Everything appears to match except that one register in check()

### CODE REVIEW

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 1]
```diff
@@ -4,118 +4,299 @@
...
     /* Parameters   */ 0,
     /* Leaf SubMtd  */ &g_fopAc_Method.base,
     /* Priority     */ 0x007C,
-    /* Actor SubMtd */ &daObjEbomzo::Mthd_Ebomzo,
-    /* Status       */ fopAcStts_NOCULLEXEC_e | fopAcStts_CULL_e | fopAcStts_UNK40000_e,
+    /* Actor SubMtd */ &daObjEbomzo::Mthd_Ebomzo, //Mthd_Ebomzo__Q211daObjEbomzo28@unnamed@d_a_obj_ebomzo_cpp@,
+    /* Status       */ fopAcStts_NOCULLEXEC_e | fopAcStts_CULL_e | fopAcStts_UNK40000_e, //0x00040180
     /* Group        */ fopAc_ACTOR_e,
     /* CullType     */ fopAc_CULLBOX_CUSTOM_e,
 };
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:302]: can you revert the changes to the actor profile so the style stays consistent across all actors?

File: include/d/actor/d_a_obj_ebomzo.h [hunk 2]
```diff
@@ -1,28 +1,52 @@
...
-        void Execute(float(**)[3][4]);
-        BOOL Draw();
-    
+
+        static Mtx M_tmp_mtx; //.bss 80810060
+        static const char M_arcname[];
+
     public:
+
         /* Place member variables here */
```
Comment (LagoLunatic) [inline include/d/actor/d_a_obj_ebomzo.h:41]: you can remove the placeholder comment

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 3]
```diff
@@ -4,118 +4,299 @@
...
+
+},
+
+/*0x30 mSphAttr*/{
+    /* > 0x30 mSph*/{
+        /* >> 0x30 mCenter*/ {0,0,0},
+        /* >> 0x3C mRadius*/ 10.0f
+    }
+}
+};
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:80]: these collision structs are nested so much that I think it's hard to read when fully indented like this, I usually go with a simplified format where it's more flat (e.g. like in d_a_nh.cpp)

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 4]
```diff
@@ -4,118 +4,299 @@
...
 }
 
 /* 0000012C-0000021C       .text Create__Q211daObjEbomzo5Act_cFv */
-s32 daObjEbomzo::Act_c::Create() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Create() {
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+    init_mtx();
+
+    fopAcM_setCullSizeBox(this, -1000.0, -100.0, -1000.0, 1000.0, 1000.0, 1000.0);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:95]:
```suggestion
    fopAcM_setCullSizeBox(this, -1000.0f, -100.0f, -1000.0f, 1000.0f, 1000.0f, 1000.0f);
```

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 5]
```diff
@@ -4,118 +4,299 @@
...
 void daObjEbomzo::Act_c::demo() {
-    /* Nonmatching */
+    mXRotRate += mXRotRate / 8;
+    current.angle.x += mXRotRate;
+
+    if (current.angle.x >= 0x4000) {
+        current.angle.x = 0x4000;
+        mMode = Ebomzo_Mode_Fall;
+
+        dComIfGp_getVibration().StartShock(4, -33, cXyz(0.0, 1.0, 0.0));
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:197]:
```suggestion
        dComIfGp_getVibration().StartShock(4, -33, cXyz(0.0f, 1.0f, 0.0f));
```

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 6]
```diff
@@ -4,118 +4,299 @@
...
 }
 
 /* 000009C0-00000AFC       .text Execute__Q211daObjEbomzo5Act_cFPPA3_A4_f */
-void daObjEbomzo::Act_c::Execute(float(**)[3][4]) {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Execute(Mtx** matrix) {
+    cXyz offset;
+
+    offset.x = cM_ssin(current.angle.y) * 50.0f;
+    offset.y = 320.0;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:220]:
```suggestion
    offset.y = 320.0f;
```

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 7]
```diff
@@ -4,118 +4,299 @@
...
 /* 00000AFC-00000B9C       .text Draw__Q211daObjEbomzo5Act_cFv */
-BOOL daObjEbomzo::Act_c::Draw() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Draw() {
+    g_env_light.settingTevStruct(TEV_TYPE_BG0, &current.pos, &tevStr);
+    g_env_light.setLightTevColorType(mpModel, &tevStr);
+    dComIfGd_setListBG();
+    mDoExt_modelUpdateDL(mpModel);
+    dComIfGd_setList();
+    dComIfG_Bgsp(); //In debug map but not used for anything??
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:246]: the debug map suggests `dComIfG_Bgsp()->DebugDrawPoly` was called here, which was probably a debug visualization function that got ifdef'd out for the retail version.

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 8]
```diff
@@ -4,118 +4,299 @@
...
+
+        dComIfGp_particle_set(0x828f, &current.pos, &current.angle);
+    }
+
 }
 
 /* 000009BC-000009C0       .text fall__Q211daObjEbomzo5Act_cFv */
 void daObjEbomzo::Act_c::fall() {
-    /* Nonmatching */
+    return;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:212]: return statement not needed here

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 9]
```diff
@@ -4,118 +4,299 @@
...
+
+    if (mCollider.ChkTgHit()) {
+        acActor = mCollider.GetTgHitAc();
+        if (acActor) {
+            diffVec = mCollider.GetC() - acActor->current.pos;
+            if (diffVec.abs() < 100.0f) {
+                mXRotRate = 8;
+                current.angle.x += mXRotRate;
+
+                fopAcM_seStartCurrent(this, 0x69ac, 0);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:170]:
```suggestion
                fopAcM_seStartCurrent(this, JA_SE_OBJ_BOMB_DN_ST_S, 0);
```

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 10]
```diff
@@ -4,118 +4,299 @@
...
-    /* Nonmatching */
+    mXRotRate += mXRotRate / 8;
+    current.angle.x += mXRotRate;
+
+    if (current.angle.x >= 0x4000) {
+        current.angle.x = 0x4000;
+        mMode = Ebomzo_Mode_Fall;
+
+        dComIfGp_getVibration().StartShock(4, -33, cXyz(0.0, 1.0, 0.0));
+        fopAcM_seStartCurrent(this, 0x69ad, 0);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:198]:
```suggestion
        fopAcM_seStartCurrent(this, JA_SE_OBJ_BOMB_DN_ST_E, 0);
```

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 11]
```diff
@@ -4,118 +4,299 @@
...
+                mXRotRate = 8;
+                current.angle.x += mXRotRate;
+
+                fopAcM_seStartCurrent(this, 0x69ac, 0);
+                mMode = Ebomzo_Mode_Demo;
+
+                fopAcM_onSwitch(this, prm_get_swSave());
+                
+                if (!mpParticleEmitter) {
+                    //color = tevStr.mColorK0;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:176]: looks like old code

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 12]
```diff
@@ -4,118 +4,299 @@
...
 }
 
 /* 000005AC-0000062C       .text set_mtx__Q211daObjEbomzo5Act_cFv */
 void daObjEbomzo::Act_c::set_mtx() {
-    /* Nonmatching */
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::ZXYrotM(shape_angle);
+
+    PSMTXCopy(mDoMtx_stack_c::now, mpModel->mBaseTransformMtx);
+    PSMTXCopy(mDoMtx_stack_c::now, M_tmp_mtx);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:145]:
```suggestion
    MTXCopy(mDoMtx_stack_c::now, mpModel->mBaseTransformMtx);
    MTXCopy(mDoMtx_stack_c::now, M_tmp_mtx);
```

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 13]
```diff
@@ -4,118 +4,299 @@
...
+    }
+}
+};
 
 /* 00000078-0000012C       .text CreateHeap__Q211daObjEbomzo5Act_cFv */
-void daObjEbomzo::Act_c::CreateHeap() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::CreateHeap() {
+    J3DModelData* model_data = (J3DModelData*)dComIfG_getObjectRes(M_arcname, 4);
+    JUT_ASSERT(140, model_data != 0);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:85]:
```suggestion
    JUT_ASSERT(140, model_data != NULL);
```

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 14]
```diff
@@ -4,118 +4,299 @@
...
+        /* >> 0x3C mRadius*/ 10.0f
+    }
+}
+};
 
 /* 00000078-0000012C       .text CreateHeap__Q211daObjEbomzo5Act_cFv */
-void daObjEbomzo::Act_c::CreateHeap() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::CreateHeap() {
+    J3DModelData* model_data = (J3DModelData*)dComIfG_getObjectRes(M_arcname, 4);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:84]: the res folder has enums for every arc's list of files, so if you include d/res/res_ebomzo.h you can avoid the magic number like this:
```suggestion
    J3DModelData* model_data = (J3DModelData*)dComIfG_getObjectRes(M_arcname, EBOMZO_BDL_EBOMZO);
```

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 15]
```diff
@@ -4,118 +4,299 @@
...
-void daObjEbomzo::Act_c::Mthd_Create() {
-    /* Nonmatching */
+int daObjEbomzo::Act_c::Mthd_Create() {
+    int phase_state;
+
+    fopAcM_SetupActor(this, daObjEbomzo::Act_c);
+    
+    phase_state = dComIfG_resLoad(&mPhs, M_arcname);
+    if (phase_state == cPhs_COMPLEATE_e) {
+        phase_state = MoveBGCreate(M_arcname, 7, NULL, 0x1200);
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:121]:
```suggestion
        phase_state = MoveBGCreate(M_arcname, EBOMZO_DZB_EBOMZO, NULL, 0x1200);
```

File: src/d/actor/d_a_obj_ebomzo.cpp [hunk 16]
```diff
@@ -4,118 +4,299 @@
...
+        s32 Mthd_Draw(void* i_this) {
+            return static_cast<Act_c*>(i_this)->Draw();
+        }
 
-/* 00000C28-00000C54       .text Mthd_IsDelete__Q211daObjEbomzo28@unnamed@d_a_obj_ebomzo_cpp@FPv */
-void Mthd_IsDelete(void*) {
-    /* Nonmatching */
-}
+        /* 00000C28-00000C54      .text Mthd_IsDelete__Q211daObjEbomzo28@unnamed@d_a_obj_ebomzo_cpp@FPv  */
+        s32 Mthd_IsDelete(void* i_this) {
```
Comment (LagoLunatic) [inline src/d/actor/d_a_obj_ebomzo.cpp:273]: other than Create, these other 4 methods in the method table return `BOOL`

File: src/d/actor/d_a_obj_ebomzo.cpp (hunk 12 above)
Comment (magcius) [reply src/d/actor/d_a_obj_ebomzo.cpp:145]: mpModel->setBaseTRMtx() and cMtx_copy, no?

Comment (LagoLunatic) [reply src/d/actor/d_a_obj_ebomzo.cpp:145]: ah yeah, you're right

File: src/d/actor/d_a_obj_ebomzo.cpp (hunk 7 above)
Comment (waffleoRai) [reply src/d/actor/d_a_obj_ebomzo.cpp:246]: Should I remove it then? For now, I just left it alone and removed the comment.
